MINOR: release.py: fix some compatibility problems.

cmccabe opened the pull request:
Rather than using sed, use built-in Python regular expressions to strip
the SNAPSHOT expression from the pom.xml files.  Sed has different flags
on different platforms, such as Linux.  Using Python directly here is
more compatible, as well as being more efficient, and not requiring an
rm command afterwards.

When running release_notes.py, use the current Python interpreter.
This is needed to prevent attempting to run release_notes.py with
Python 3 on some systems.  release_notes.py will not (yet) work with
Python 3.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

edenhill commented:
> When running release_notes.py, use the current Python interpreter.
This is needed to prevent attempting to run release_notes.py with
Python 3 on some systems. release_notes.py will not (yet) work with
Python 3.

Alternatively change the release_notes.py shebang to: `#!/usr/bin/env python2`

edenhill reviewed (approved):
LGTM

cmccabe commented:
> Alternatively change the release_notes.py shebang to: #!/usr/bin/env python2

Unfortunately some platforms, like MacOS, don't provide a python2 symlink.

cmccabe committed:
MINOR: release.py: fix some compatibility problems

Rather than using sed, use built-in Python regular expressions to strip
the SNAPSHOT expression from the pom.xml files. Sed has different flags
on different platforms, such as Linux. Using Python directly here is
more compatible, as well as being more efficient, and not requiring an
rm command afterwards.

When running release_notes.py, use the current Python interpreter.
This is needed to prevent attempting to run release_notes.py with
Python 3 on some systems. release_notes.py will not (yet) work with
Python 3.

mumrah reviewed (commented):
I would keep the print statements as is done in the `cmd` function. Helps trace progress which could be useful in figuring out where things went wrong (in case of an error).

cmccabe committed:
Add print statements

cmccabe commented:
Good point.  I added back the print statements.

omkreddy reviewed (approved):
Thanks for the PR. LGTM.

mumrah reviewed (approved):
LGTM :shipit:
